src/h3/api/basic_int/__init__.py

```diff
@@ -564,30 +570,27 @@ def h3shape_to_cells_experimental(h3shape, res, flags=0):
     There is currently no guaranteed order of the output cells.
     """
 
-    if isinstance(flags, str):
-        try:
-            flags = ContainmentMode[flags]
-        except KeyError as e:
-            raise ValueError('Unrecognized flags: ' + flags) from e
-    if isinstance(flags, ContainmentMode):
-        flags = int(flags)
-    if not isinstance(flags, int):
-        raise ValueError(
-            'Flags should be ContainmentMode, str, or int, but got: ' + str(type(flags))
-        )
+    containment_modes = {
+        'center': 0,
+        'full': 1,
+        'overlap': 2,
+        'bbox_overlap': 3,
+    }
```

Comment:
Thoughts on names? I took a stab here to shorten things, but we should also align with upstream. I could go either way with `bbox_overlap` vs `overlap_bbox`, but this way each containment mode has a distinct first letter and should be less likely to accidentally select `overlap_bbox` when going for `overlap`.